transcom/mymove
# B-21427 & B-21430: Update & Populate Addresses is_oconus value

## [B-21427](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21427)
## [B-21430](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21430)

## Summary
This PR includes the following changes:

1. SQL script to update address records in the addresses table with a is_oconus value
2. Populating the is_oconus value on the addresses table when an address is created
3. Populating the is_oconus value on the addresses table when an address is updated

*At this time AK and HI is considered OCONUS

There are many areas throughout the mymove workflow and office workflow where an address is created and updated. It would be good to go through all workflows and make sure places where an address is updated or created works properly. is_oconus is not nullable so if an address is being created or updated without an is_oconus value it will throw an error.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

*Note: AK and HI is considered OCONUS
### How to test migration file
1. Run the migrations
2. Verify that the record in the addresses table have a true or false value for is_oconus
3. If country is US or United States then is_oconus should be false. Anything, including state being AK or HI should make is_oconus true

### How to test on address creation
1. Log into MilMove as a customer
2. Create a new shipment and add the address information
3. Check the addresses table for the address you created and make sure is_oconus value is populated

### How to test on service item creation
1. Log in as Prime
2. Go to a shipment
3. Create a serviceItem on a shipment and add an address
5. Check the addresses table for the address you entered and make sure is_oconus value is populated

### How to test address update
1. Log in as Prime
2. Go to a move
3. Update shipments destination address on a shipment and add an address
5. Check the addresses table for the address you entered and make sure is_oconus value is populated


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
